frontend/packages/console-dynamic-plugin-sdk/src/shared-modules.ts

```diff
@@ -15,4 +35,26 @@ export const sharedPluginModules = [
   'react-redux',
   'redux',
   'redux-thunk',
-];
+] as const;
+
+/**
+ * Metadata associated with the shared modules.
+ */
+export const sharedPluginModulesMetadata: Record<
+  typeof sharedPluginModules[number],
```

Comment:
Since `sharedPluginModules` is now a readonly const array, above type parameter will resolve to
```ts
"@openshift-console/dynamic-plugin-sdk" | ... | "redux-thunk"
```
which means `sharedPluginModulesMetadata` will be kept in-sync with `sharedPluginModules` items.